lib/class-wp-rest-menus-controller.php

```diff
@@ -305,6 +381,40 @@ public function delete_item( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Updates the menu's locations from a REST request.
+	 *
+	 * @param int             $menu_id The menu id to update the location form.
+	 * @param WP_REST_Request $request The request object with menu and locations data.
+	 *
+	 * @return null|WP_Error WP_Error on an error assigning any of the locations, otherwise null.
+	 */
+	protected function handle_locations( $menu_id, $request ) {
+		if ( ! isset( $request['locations'] ) ) {
+			return true;
+		}
+
+		$menu_locations = get_registered_nav_menus();
+		$menu_locations = array_keys( $menu_locations );
+		$new_locations  = array();
+		foreach ( $request['locations'] as $location ) {
+			if ( ! in_array( $location, $menu_locations, true ) ) {
+				return new WP_Error( 'none_exist_location', __( 'Menu location does not exist.' ), array( 'status' => 400 ) );
+			}
+			$new_locations[ $location ] = $menu_id;
+		}
+		$assigned_menu = get_nav_menu_locations();
+		foreach ( $assigned_menu as $location => $term_id ) {
+			if ( $term_id === $menu_id ) {
+				unset( $assigned_menu[ $location ] );
+			}
+		}
+		$new_assignments = array_merge( $new_locations, $assigned_menu );
```

Comment:
I think this `array_merge()` is backwards.

This test is failing for me.

```php
	public function test_change_menu_location() {
		$this->register_nav_menu_locations( array( 'primary', 'secondary' ) );
		$secondary_id = self::factory()->term->create(
			array(
				'name'        => 'Secondary Name',
				'description' => 'Secondary Description',
				'slug'        => 'secondary-slug',
				'taxonomy'    => 'nav_menu',
			)
		);

		$locations              = get_nav_menu_locations();
		$locations['primary']   = self::$term_id;
		$locations['secondary'] = $secondary_id;
		set_theme_mod( 'nav_menu_locations', $locations );

		wp_set_current_user( self::$administrator );

		$request = new WP_REST_Request( 'PUT', '/wp/v2/menus/' . self::$term_id );
		$request->set_body_params(
			array(
				'locations' => array( 'secondary' ),
			)
		);
		$response = rest_get_server()->dispatch( $request );

		$this->assertEquals( 200, $response->get_status() );

		$locations = get_nav_menu_locations();
		$this->assertArrayNotHasKey( 'primary', $locations );
		$this->assertArrayHasKey( 'secondary', $locations );
		$this->assertEquals( self::$term_id, $locations['secondary'] );
	}
```

Reply:
Fixed.